From 4de0c17f32e3779b50d409eb750ce1cb9e9ee7d9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 16 May 2019 21:26:55 +0200
Subject: [RFC/PATCH 1/4] fsck: test for MAXINT + 1, not + 2 in date oveflow
 test
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change a test added in d4b8de0420 ("fsck: report integer overflow in
author timestamps", 2014-02-24) to test for the maximum 64 bit
unsigned integer + 1, not +2. Having this off-by-one was confusing,
even though the test served its purpose.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/t1450-fsck.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/t/t1450-fsck.sh b/t/t1450-fsck.sh
index 0f268a3664..73ab0a8ee9 100755
--- a/t/t1450-fsck.sh
+++ b/t/t1450-fsck.sh
@@ -196,10 +196,10 @@ test_expect_success '> in name is reported' '
 	test_i18ngrep "error in commit $new" out
 '
 
-# date is 2^64 + 1
+# date is 2^64
 test_expect_success 'integer overflow in timestamps is reported' '
 	git cat-file commit HEAD >basis &&
-	sed "s/^\\(author .*>\\) [0-9]*/\\1 18446744073709551617/" \
+	sed "s/^\\(author .*>\\) [0-9]*/\\1 18446744073709551616/" \
 		<basis >bad-timestamp &&
 	new=$(git hash-object -t commit -w --stdin <bad-timestamp) &&
 	test_when_finished "remove_object $new" &&
-- 
2.21.0.1020.gf2820cf01a

